packages/date-picker/src/vaadin-month-calendar.js

```diff
@@ -340,36 +365,54 @@ class MonthCalendar extends ThemableMixin(PolymerElement) {
     e.preventDefault();
   }
 
-  _getRole(date) {
-    return date ? 'button' : 'presentation';
+  __getWeekNumber(days) {
+    const date = days.reduce((acc, d) => {
+      return !acc && d ? d : acc;
+    });
+
+    return getISOWeekNumber(date);
   }
 
-  _getAriaLabel(date) {
-    if (!date) {
-      return '';
-    }
+  __isDayFocused(date, focusedDate) {
+    return dateEquals(date, focusedDate);
+  }
+
+  __isDaySelected(date, selectedDate) {
+    return dateEquals(date, selectedDate);
+  }
 
-    var ariaLabel =
-      this._getDate(date) +
-      ' ' +
-      this.i18n.monthNames[date.getMonth()] +
-      ' ' +
-      date.getFullYear() +
-      ', ' +
-      this.i18n.weekdays[date.getDay()];
-
-    if (this._isToday(date)) {
-      ariaLabel += ', ' + this.i18n.today;
+  __getDayAriaSelected(date, selectedDate) {
+    if (this.__isDaySelected(date, selectedDate)) {
+      return 'true';
     }
+  }
 
-    return ariaLabel;
+  __isDayDisabled(date, minDate, maxDate) {
+    return !dateAllowed(date, minDate, maxDate);
   }
 
-  _getAriaDisabled(date, min, max) {
+  __getDayAriaDisabled(date, min, max) {
     if (date === undefined || min === undefined || max === undefined) {
       return;
     }
-    return this._dateAllowed(date, min, max) ? 'false' : 'true';
+
+    if (this.__isDayDisabled(date, min, max)) {
+      return 'true';
+    }
+  }
+
+  __getDayTabindex(date, focusedDate) {
+    if (this.__isDayFocused(date, focusedDate)) {
+      return '0';
+    }
+
+    return '-1';
```

Comment:
nit: untested (ignore, in case the tests are to be added with the following refactors)

Reply:
Thanks. I will enhance keyboard navigation tests in the next PR.